aurawatch.lua

```diff
@@ -476,6 +507,7 @@ function AuraWatch:PLAYER_ENTERING_WORLD()
 	end
 end
 
+local filters = {"HELPFUL", "HARMFUL"}
```

Comment:
Why break this out?

Reply:
As I mentioned in the commit title, it avoids creating a table every time UNIT_AURA is triggered which can be a lot, so it saves memory. Lua doesn't have any auto optimization for these things and repeatedly allocating new tables may slow things down. Probably not that big of a deal but thought i'd just move it outside the function.

Reply:
I was genuinely curious and put together a simple benchmark...

```
local iterations = 10000000
local filters = {"HARMFUL", "HELPFUL"}

function test_loop_with_global_hash()
  for _, filter in pairs(filters) do
  end
end

function  test_loop_with_local_hash()
  for _, filter in pairs({"HELPFUL", "HARMFUL"}) do
  end
end


for i = 1, iterations do
  -- test_loop_with_local_hash()
  test_loop_with_global_hash()
end
```

results with only `test_loop_with_global_hash`:
```
time lua luatest.lua

real    0m1.536s
user    0m1.510s
sys     0m0.003s

```

results with only `test_loop_with_local_hash`:
```
time lua luatest.lua

real    0m2.491s
user    0m2.396s
sys     0m0.004s
```

In 10M iterations it does make a difference in CPU time and surely memory as well, but didn't analyze that.

Reply:
Interesting, I've never actually profiled it myself, just gone by what I have read. I think the most impact would be on memory though, especially since wow's garbage collector doesn't run in combat unless max memory is reached. Imagine a 10min arena game without leaving combat once, that *could* be a lot of memory wasted, and if the GC starts running in combat you will get FPS spikes. This would obviously not happen with this filter table alone but if several addons does it you'll reach the threshold for running GC very quick.

However, it sounds like Blizzard actually has been recently doing some performance optimizations to their garbage collector. [Interesting read about it here in the comments by Simca](http://www.wowinterface.com/forums/showthread.php?t=56290&page=2).

Point of interest are:
> They've greatly improved garbage collection performance recently, and have monitored the performance involved in using tons of small, disposable tables under the new GC implementation. It is an incredibly small performance hit.

> the optimizations for the garbage collector he is talking about were changes that went Live with Patch 7.0.3.

So it seems like we don't really have to worry about these things anymore 👍

Reply:
Very good to know!